### PR TITLE
Stream yarn/npm output

### PR DESCRIPTION
**Pull request by Rowno**
This adds yarn/npm output streaming so that you can see the real time install output / progress bar. This also makes it so that the yarn/npm output colors are maintained. I also [added execa for these reasons](https://github.com/sindresorhus/execa#why).

PS: you should remove prettier from the devDependencies since it's misleading.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -43,12 +43,13 @@ if (!YARNHOOK_BYPASS) {
     console.log("CMD:", CMD);
   }
 
-  if (lockfile !== null) {
+  if (lockfile) {
```
Comment (frontsideair) [inline index.js]: Is there a reason you changed this? If I'm not wrong it can only be "npm", "yarn", or `null`.

File: index.js [hunk 2]
```diff
@@ -43,12 +43,13 @@ if (!YARNHOOK_BYPASS) {
...
     // run a git diff on the lockfile
-    const output = execSync(`git diff HEAD@{1}..HEAD@{0} -- ${lockfile}`, {
-      cwd: gitDir,
-      encoding: "utf-8"
-    });
+    const { stdout: output } = execa.sync(
```
Comment (frontsideair) [inline index.js]: Does it always return a string? The docs say the return value is same as [child_process.spawnSync](https://nodejs.org/api/child_process.html#child_process_child_process_spawnsync_command_args_options), which is documented as `<Buffer> | <string>`.

**Comment by frontsideair** [comment]
Thanks for the PR!

I'm curious why do you think it's misleading that prettier is a devDependency? I'm under the impression that linters, transpilers, bundlers etc. are fine in devDependencies. Even though it's not enforced yet I expect the project code to be prettified for consistent style.

File: index.js (hunk 2 above)
Comment (frontsideair) [reply index.js]: I tested this and I think its length is never zero. I wonder how it decides to return either Buffer or string.

File: index.js (hunk 1 above)
Comment (Rowno) [reply index.js]: I figured it would be better to check if it's truthy. I can change it back if you want.

**Comment by Rowno** [comment]
It's misleading because the javascript hasn't been run through prettier at all. When I saved the file with prettier enabled the entire file changed. So I assume it isn't being used which is misleading.

File: index.js (hunk 2 above)
Comment (Rowno) [reply index.js]: Yep, I tested it and it seems to work fine.

File: index.js (hunk 1 above)
Comment (frontsideair) [reply index.js]: I'd like to keep the PR only to relevant stuff, if you don't mind. 😊

**Comment by frontsideair** [comment]
I just ran prettier on the current code and it didn't change. Maybe your editor config or some global settings interfered?

**Comment by Rowno** [comment]
I think it's because you don't have any prettier config defined the in the `package.json` or a `. prettierrc` file, so it's falling back to my defaults.

**Comment by frontsideair** [comment]
I think you're right, I'll add a default prettier config and start enforcing it.

**Comment by frontsideair** [comment]
Just did it, you can rebase now.

**Comment by frontsideair** [comment]
Merged, thanks a bunch! 👍

**Pull request merged**